omero_figure/views.py

```diff
@@ -116,6 +123,83 @@ def img_data_json(request, image_id, conn=None, **kwargs):
     return HttpResponse(json.dumps(rv), content_type='json')
 
 
+@login_required()
+def render_scaled_region(request, z, t, iid, conn=None, **kwargs):
+
+    region = request.GET.get('region')
+    logger.debug("Rendering region: %s, Image: %s" % (region, iid))
+
+    x, y, width, height = [float(r) for r in region.split(',')]
+    max_size = request.GET.get('max_size', 2000)
+    max_size = int(max_size)
+
+    pi = _get_prepared_image(request, iid, conn=conn)
+    if pi is None:
+        raise Http404
+    image, compress_quality = pi
+
+    size_x = image.getSizeX()
+    size_y = image.getSizeY()
+
+    scale_levels = image.getZoomLevelScaling()
+    if scale_levels is None:
+        # Not a big image - can load at full size
+        level = None
+    else:
+        # Pick zoom such that returned image is below MAX size
+        max_level = len(scale_levels.keys()) - 1
+        longest_side = max(width, height)
+
+        # start small, and go until we reach target size
+        zm = max_level
+        while zm > 0 and scale_levels[zm - 1] * longest_side < max_size:
+            zm = zm - 1
+
+        level = max_level - zm
+
+        # We need to use final rendered jpeg coordinates
+        # Convert from original image coordinates by scaling
+        scale = scale_levels[zm]
+        x = int(x * scale)
+        y = int(y * scale)
+        width = int(width * scale)
+        height = int(height * scale)
+        size_x = int(size_x * scale)
+        size_y = int(size_y * scale)
+
+    canvas = None
+    # Coordinates below are all final jpeg coordinates & sizes
+    if x < 0 or y < 0 or (x + width) > size_x or (y + height) > size_y:
+        # If we're outside the bounds of the image...
+        # Need to render reduced region and paste on to full size image
+        canvas = Image.new("RGBA", (width, height), (221, 221, 221))
+        paste_x = 0
+        paste_y = 0
+        if x < 0:
+            paste_x = -x
+            width = width + x
+            x = 0
+        if y < 0:
+            paste_y = -y
+            height = height + y
+            y = 0
+
+    # Render the region...
+    jpeg_data = image.renderJpegRegion(z, t, x, y, width, height, level=level,
```

Comment:
why not using ``render_image_region``?

Reply:
Do you mean ```image.renderImageRegion()```? That doesn't exist.
```image.renderImage()``` returns a PIL image and we mostly want the ```jpeg_data``` to return in the response.

Reply:
maybe too much logic in this PR that it cannot be done